wordpress-mobile/WordPress-iOS
# Feature/exporter progress report

Fixes #8308 

This PR adds progress reporting and the possibility of cancelation during the export of media from the Photo library to the app.

This is specially important when using iCloud media, where the media may need to be downloaded before being used and on Video insertion where the export process can take the same time has the upload.

This was implemented by making all the exporters reporting their progress and hooking this information with the `createMedia` method on `MediaService` and then connecting this to the addMedia method on the `MediaCoordinator`.

To test:
 - Open the Media Library of a Site
 - Tap to add new media
 - Select the Async option
 - Select media that is stored on iCloud or/and that is a large video
 - See if the initial part display progress.
